pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -812,6 +775,274 @@ private void createDerivedColumnV1Indices(String column, FunctionEvaluator funct
     }
   }
 
+  /**
+   * Helper method to convert the output of a transform function to the appropriate type for an SV or MV
+   * {@link FieldSpec.DataType#INT} field
+   *
+   * @param outputValue the output of the transform function
+   * @param isSingleValue true if the field (column) is single-valued
+   * @param outputValueType the output value type for the transform function; can be null (in which case,
+   *                        the {@code outputValue} should be the field's default null value)
+   * @param defaultNullValue the default null value for the field
+   * @param dictionary true if the column has a dictionary. For an MV field, this results in a primitive array being
+   *                   returned rather than an array of the primitive wrapper class
+   * @return the converted output value (either an Integer, an Integer[] or an int[])
+   */
+  private Object getIntOutputValue(Object outputValue, boolean isSingleValue, PinotDataType outputValueType,
+      Integer defaultNullValue, boolean dictionary) {
```

Comment:
Pass in primitive `int` for better performance. Same for other methods

Reply:
The default null value obtained from the field spec will be of the primitive object wrapper class type, and the value that we're returning here is also an `Object` since that's the type for the `outputValues` array. Won't using a primitive `int` here simply result in a redundant additional autoboxing and unboxing?

Reply:
Good point. Only one code path is using the primitive type, so I guess we can keep the boxed value. In the future we may consider changing the object array to primitive array to save memory, but that is out of the scope of this PR